CRM/Upgrade/Incremental/php/FiveSixtyTwo.php

```diff
@@ -23,6 +23,19 @@ class CRM_Upgrade_Incremental_php_FiveSixtyTwo extends CRM_Upgrade_Incremental_B
 
   public function setPreUpgradeMessage(&$preUpgradeMessage, $rev, $currentVer = NULL) {
     if ($rev == '5.62.alpha1') {
+
+      // Copied from FiveFiftySeven, only display if we upgrading from version after 5.57.alpha1
+      if (version_compare($currentVer, '5.57.alpha1', '>')) {
+        $docUrl = 'https://civicrm.org/redirect/activities-5.57';
+        $docAnchor = 'target="_blank" href="' . htmlentities($docUrl) . '"';
+        if (CRM_Core_DAO::singleValueQuery('SELECT COUNT(id) FROM civicrm_activity WHERE is_current_revision = 0')) {
+          // Text copied from FiveFifty Seven
+          $preUpgradeMessage .= '<p>' . ts('Your database contains CiviCase activity revisions which are deprecated and will begin to appear as duplicates in SearchKit/api4/etc.<ul><li>For further instructions see this <a %1>Lab Snippet</a>.</li></ul>', [1 => $docAnchor]) . '</p>';
+          // New text explaination as to why we show this again.
+          $preUpgradeMessage .= '<p>' . ts('Note: You might have followed these steps already but unfortunately a previous upgrade which removed CiviCase Activity revisions failed to account for all the places this functionality existed. This means that either new Revisions have been added since you removed them, or not all were removed.') . '</p>';
```

Comment:
I still think the word "removed" might cause some confusion - nothing was removed by the upgrade. If you want to keep the word "removed" then how about "...a previous upgrade which told you to remove CiviCase Activity revisions..."?

Reply:
sorry yes totally will address.